jsonrpc/namespaces/eth/filters/api.go

```diff
@@ -0,0 +1,679 @@
+// Copyright 2021 Evmos Foundation
+// This file is part of Evmos' Ethermint library.
+//
+// The Ethermint library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Ethermint library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Ethermint library. If not, see https://github.com/evmos/ethermint/blob/main/LICENSE
+package filters
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client"
+
+	"cosmossdk.io/log"
+
+	coretypes "github.com/cometbft/cometbft/rpc/core/types"
+	rpcclient "github.com/cometbft/cometbft/rpc/jsonrpc/client"
+	tmtypes "github.com/cometbft/cometbft/types"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/eth/filters"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/initia-labs/minievm/jsonrpc/backend"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	evmtypes "github.com/initia-labs/minievm/x/evm/types"
+)
+
+// FilterAPI gathers
+type FilterAPI interface {
+	//TODO: mid-priority
+	// NewPendingTransactionFilter() rpc.ID
+	NewBlockFilter() rpc.ID
+	NewFilter(criteria filters.FilterCriteria) (rpc.ID, error)
+	GetFilterChanges(id rpc.ID) (interface{}, error)
+	GetFilterLogs(ctx context.Context, id rpc.ID) ([]*ethtypes.Log, error)
+	UninstallFilter(id rpc.ID) bool
+	GetLogs(ctx context.Context, crit filters.FilterCriteria) ([]*ethtypes.Log, error)
+}
+
+// Backend defines the methods requided by the PublicFilterAPI backend
+type Backend interface {
```

Comment:
@jasonsong0 
Do we need this? What's the purpose of Backend interface?
Compile is successful without this interface.

Reply:
it is not used. i will delete it